torch/_decomp/decompositions.py

```diff
@@ -4114,6 +4114,53 @@ def upsample_bicubic2d_vec(
     return upsample_bicubic2d_default(a, output_size, align_corners, scale_h, scale_w)
 
 
+@register_decomposition(aten.reflection_pad1d)
+@register_decomposition(aten.reflection_pad2d)
+@register_decomposition(aten.reflection_pad3d)
+@pw_cast_for_opmath
+@out_wrapper()
+def _reflection_pad(a: Tensor, padding: Tuple[int, ...]) -> Tensor:
+    dim = len(padding) // 2
+    torch._check(
+        a.dim() in (dim + 1, dim + 2),
+        lambda: f"reflection_pad{dim}d requires {dim + 1}D or {dim + 2}D input",
+    )
+    inp_shape = a.shape[-dim:]
+    nc_dim = a.dim() - dim
+
+    padding_left = [padding[2 * (dim - 1 - i)] for i in range(dim)]
+    padding_right = [padding[2 * (dim - 1 - i) + 1] for i in range(dim)]
+
+    result = a
+    for i in range(dim):
+        middle = torch.arange(0, inp_shape[i], step=1, device=a.device)
+        if padding_left[i] < 0:
+            middle = middle[-padding_left[i] :]
+            to_cat = [middle]
+        else:
+            left = torch.arange(padding_left[i], 0, step=-1, device=a.device)
+            to_cat = [left, middle]
+
+        if padding_right[i] < 0:
+            middle = middle[: -padding_right[i]]
+        else:
+            right = torch.arange(
+                inp_shape[i] - 2,
+                inp_shape[i] - 2 - padding_right[i],
+                step=-1,
+                device=a.device,
+            )
+            to_cat.append(right)
+        idx: List[Any] = [slice(s) for s in result.shape]
+        idx[i + nc_dim] = torch.cat(tuple(to_cat))
+        result = result[idx]
```

Comment:
We should use `_unsafe_index` here which skips bounds checking, same as `check=False` which is used in the lowering.
```suggestion
        result = aten._unsafe_index(result, idx)
```

Reply:
Thanks